lib/between_meals/knife.rb

```diff
@@ -73,6 +74,18 @@ def cookbook_upload_all
       exec!("#{@knife} cookbook upload -a -c #{@config}", @logger)
     end
 
+    def berks_cookbook_upload_all
+      @cookbook_dirs.each do |path|
+        Dir.chdir(path)  # http://stackoverflow.com/a/808285/830426
+        cookbook_dirs=Dir["*"].reject{|o| not File.directory?(o)}
+        cookbook_dirs.each do |dir|
+          @logger.warn("Running berkshelf on cookbook: #{dir}")
+          exec!("cd #{path}/#{dir} && #{@berks} install && #{@berks} upload", 
```

Comment:
I'm surprised this `cd` worked given you had already changed the cwd of GD before spawning the subshell...